src/ResourceManagement/Compute/Compute.Tests/VMScaleSetTests/VMScaleSetNetworkProfileTests.cs

```diff
@@ -0,0 +1,96 @@
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using Microsoft.Azure.Management.Resources;
+using Microsoft.Azure.Management.Resources.Models;
+using Microsoft.Azure.Management.Storage;
+using Microsoft.Azure.Management.Storage.Models;
+using Microsoft.Rest.Azure;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Text;
+using Xunit;
+
+namespace Compute.Tests
+{
+    public class VMScaleSetNetworkProfileTests : VMScaleSetTestsBase
+    {
+        /// <summary>
+        /// Associates a VMScaleSet to an Application gateway
+        /// </summary>
+        [Fact]
+        public void TestVMScaleSetWithApplciationGateway()
+        {
+            using (MockContext context = MockContext.Start(this.GetType().FullName))
+            {
+                EnsureClientsInitialized(context);
+
+                ImageReference imageRef = GetPlatformVMImage(useWindowsImage: true);
+
+                // Create resource group
+                string rgName = TestUtilities.GenerateName(TestPrefix) + 1;
+                var vmssName = TestUtilities.GenerateName("vmss");
+                string storageAccountName = TestUtilities.GenerateName(TestPrefix);
+                VirtualMachineScaleSet inputVMScaleSet;
+
+                bool passed = false;
+                try
+                {
+                    var storageAccountOutput = CreateStorageAccount(rgName, storageAccountName);
+                    var vnetResponse = CreateVNETWithSubnets(rgName, 2);
+                    var gatewaySubnet = vnetResponse.Subnets[0];
+                    var vmssSubnet = vnetResponse.Subnets[1];
+                    ApplicationGateway appgw = CreateApplicationGateway(rgName, gatewaySubnet);
+                    Microsoft.Azure.Management.Compute.Models.SubResource backendAddressPool = new Microsoft.Azure.Management.Compute.Models.SubResource()
+                    {
+                        Id = appgw.BackendAddressPools[0].Id
+                    };
+
+                    VirtualMachineScaleSet vmScaleSet = CreateVMScaleSet_NoAsyncTracking(
+                        rgName: rgName,
+                        vmssName: vmssName,
+                        storageAccount: storageAccountOutput,
+                        imageRef: imageRef,
+                        inputVMScaleSet: out inputVMScaleSet,
+                        vmScaleSetCustomizer:
+                            (virtualMachineScaleSet) =>
+                                virtualMachineScaleSet.VirtualMachineProfile.NetworkProfile
+                                    .NetworkInterfaceConfigurations[0].IpConfigurations[0]
+                                    .ApplicationGatewayBackendAddressPools.Add(backendAddressPool),
+                        createWithPublicIpAddress: false,
+                        subnet: vmssSubnet);
+
```

Comment:
Could you please add a verification for the returned vmScaleSet object?  You can use ValidateVMScaleSet() function.  Also, the current ValidateVMScaleSet() does not have Network Profile verification, so it would be better if you add that part to ValidateVMScaleSet().  Thanks.